internal/controller/projectdevelopmentstream_controller.go

```diff
@@ -180,9 +182,40 @@ func (r *ProjectDevelopmentStreamReconciler) setProductOwnerRef(ctx context.Cont
 	return r.Client.Update(ctx, pds)
 }
 
+// Returns a handler for collecting all dev streams that exist on the same namespace as
+// the object passed to the handler
```

Comment:
Isn't this a little excessive? Shouldn't we just return the PDST that the PDS is referring to, if any? This might trigger too many unnecessary reconciliation.

Reply:
The object is not necessarily a template. It could be a project as well.
I think the alternative will be to write different code for project and for template, and in each, iterate over the streams and collect the ones which belong to the relevant project or template.

I think it will be a lot of code (comparing to what we have at the moment) for optimizing something that:
1.  I'm not sure is going to happen too often (how often projects and templates are going to be modified? it's something that users need to do manually).
2. Not sure how much different it's going to make (how many templates/projects a typical tenant will have).

I think we might deffer it for later and try to assess over time whether the clutter/maintenance outweighs performance impact.

What do you think?

Reply:
ok